packages/eslint-plugin-ckeditor5-rules/tests/use-require-for-debug-mode-imports.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * @license Copyright (c) 2003-2022, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+const RuleTester = require( 'eslint' ).RuleTester;
+
+const ruleTester = new RuleTester();
+const usingImportNotAllowed = { message: 'Using import with "@if CK_DEBUG" keyword is not allowed. Please Use require().' };
+
+ruleTester.run(
+	'eslint-plugin-ckeditor5-rules/use-require-for-debug-mode-imports',
+	require( '../lib/rules/use-require-for-debug-mode-imports' ),
+	{
+		invalid: [
+			{
+				code: '// @if CK_DEBUG // import TestPackage from \'test-package\';',
+				output: '// @if CK_DEBUG // const TestPackage = require(\'test-package\');',
+				errors: [ usingImportNotAllowed ]
+			},
+			{
+				code: '// @if CK_DEBUG // import { testFunction } from \'test-package\';',
+				output: '// @if CK_DEBUG // const { testFunction } = require(\'test-package\');',
+				errors: [ usingImportNotAllowed ]
+			},
+			{
+				code: '/**\n' +
+				'* @if CK_DEBUG // import TestPackage from \'test-package\';\n' +
+				'*/',
+				output: '/**\n' +
+				'* @if CK_DEBUG // const TestPackage = require(\'test-package\');\n' +
+				'*/',
+				errors: [ usingImportNotAllowed ] }
+		],
+		valid: [
+			{
+				code: '// @if CK_DEBUG // const TestPackage = require(\'test-package\');'
+			},
+			{
+				code: '// @if CK_DEBUG // const { testFunction } = require(\'test-package\');'
+			},
+			{
+				code: '/**\n' +
+				'* @if CK_DEBUG // const TestPackage = require(\'test-package\');\n' +
+				'*/'
+			}
+		]
+	}
```

Comment:
I would see more tests as developers to ensure the rule works as expected.

1. `CK_DEBUG` is a generic flag. The testing environment supports also _named_ flags:
	* [`CK_DEBUG_MINIMAP`](https://github.com/ckeditor/ckeditor5/blob/30286f77b39526fce2856b03b9be0ba4cc91d1c7/packages/ckeditor5-minimap/src/minimap.js#L22)
	* [`CK_DEBUG_ENGINE`](https://github.com/ckeditor/ckeditor5/blob/a1f8243e0bdfedf3e71d91e64c390c8f3873086f/packages/ckeditor5-engine/src/model/operation/attributeoperation.ts#L207)
2. I quote [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import) - there are many import statements we should check, and if possibly, try to fix.
	* **An import of a default export module.**
		Invalid: 
		```js
		import defaultExport from "module-name";
		```
		Valid: 
		```js
		const defaultExport = require( "module-name" ).default;
		```
	* **An import of everything with a prefixed scope.**
		Invalid: 
		```js
		import * as name from "module-name";
		```
		Valid: 
		```js
		const name = require( "module-name" );
		```
	* **An import of a single, non-default module.**
		Invalid: 
		```js
		import { export1 } from "module-name";
		```
		Valid: 
		```js
		const { export1 } = require( "module-name" );
		```
		⚠️ We should also check if the import contains more modules. 
		All of them should be listed as `const { ... }`.
	* **An import of a default export module using brackets.**
		Invalid: 
		```js
		import { default as alias } from "module-name";
		```
		Valid: 
		```js
		const defaultExport = require( "module-name" ).default;
		```
	* **An import of a module under a changed name (using `as`).**
		Invalid: 
		```js
		import { export1 as alias1 } from "module-name";
		```
		Valid: 
		```js
		const { export1: alias1 ) = require( "module-name" );
		```

	* **Edge case: an import of non-module, e.g., a specific CSS loaded only for debugging purposes.**
		Invalid: 
		```js
		import "module-name";
		```
		Valid: 
		```js
		require( "module-name" );
		```
		I am not sure what is the expected statement. It needs to be verified manually.


The last question I have is – is the rule working with TypeScript files? I know that `import type ...` statements might fail, but we can safely ignore it as `// eslint-disable-line unused-var` will handle the problem. Types are not included in the output JavaScript files.